### ISSUE TITLE
`no-extraneous-import` should not complain about self referencing

### ISSUE THREAD

**Issue by RebeccaStevens** [mysticatea/eslint-plugin-node#333]
```json
{
  "name": "my-package",
  "type": "module"
}
```

```js
import foo from "my-package"; // This should not cause a rule violation
```
